jenkins/data-prepper/Jenkinsfile

```diff
@@ -0,0 +1,55 @@
+lib = library(identifier: 'jenkins@20211123', retriever: legacySCM(scm))
+
+pipeline {
+    agent {
+        docker {
+            label 'Jenkins-Agent-al2-x64-c54xlarge-Docker-Host'
+            image 'opensearchstaging/ci-runner:ubuntu2004-x64-docker-buildx0.6.3-qemu5.0-awscli1.22-jdk14'
+            alwaysPull true
+        }
+    }
+    parameters {
+        string(name: 'VERSION', description: 'The version of Data Prepper', trim: true)
+        string(name: 'BRANCH', description: 'Can be: <branchName>, <commitId>, refs/tags/<tagName>', trim: true)
+    }
+    stages {
+        stage('Build Data Prepper') {
+            steps {
+                script {
+                    checkout changelog: false, poll: false, scm:
+                        [$class: 'GitSCM', branches: [[name: BRANCH]], userRemoteConfigs: [[url: 'https://github.com/opensearch-project/data-prepper.git']]]
```

Comment:
I think this can be simplified to `git url: 'https://github.com/opensearch-project/data-prepper.git', branch: BRANCH`

https://github.com/opensearch-project/opensearch-build/blob/main/jenkins/opensearch/distribution-build.jenkinsfile#L44

Reply:
According to the Jenkins documentation, the `git` directive cannot work from commit ids or tags. I think in the future, we can probably use the latest of a branch, but for now we do need to build from a specific commit.

Reply:
Good to know!